components/modelregistry/modelregistry.go

```diff
@@ -82,6 +85,44 @@ func (m *ModelRegistry) ReconcileComponent(_ context.Context, cli client.Client,
 				return fmt.Errorf("failed to update image from %s : %w", Path, err)
 			}
 		}
+
+		// Create model registry viewer and editor ClusterRoles if not present
```

Comment:
any reason not add the clusterrole in model registery part either as manifests or logic, instead of having it here?

Reply:
There is nothing dynamic or dependent on the DSC/DSCI config in this so we could keep it under kustomize umbrella. I agree this fits better in `overlays/odh` if it's ODH-specific.

Reply:
Basically, these roles have the same lifecycle requirements for these clusterroles as the shared namespace in #930.

Reply:
Let's blame it for lack of coffee :) It was early for me. Maybe worth adding a comment around this piece as of why it's here, not in the manifest. So some other lost soul like me wouldn't waste ppls time asking questions.

Reply:
But I wonder - couldn't Model Registry operator take care of the needed roles instead?

Reply:
True, that was an option I considered. But, I figured these were a user facing/odh platform concern and hence better created in the odh operator instead.

Reply:
all in all, the reason to creata 2  clusterroles  is that they should not be removed when model registry component is set to Removed, is this correct?
two ways i can think of making this work:
- use manifests from Model Registry +  filter out these 2 in ODH operator to not "delete"
- creating them in ODH operator (what this PR is doing)

Reply:
when i say filter out:
- either we do the hard code filter out like how we treated some resources like dashboard's CRD
- or gather the resources you do not want to delete (i.e these 2 into one decidated folder under kustomize), thenn ODH operator call `DeployManifestsFromPath(...., true)` on this path